medmodels/statistic_evaluations/statistical_analysis/concepts_analysis.py

```diff
@@ -0,0 +1,66 @@
+"""Module for analyzing concepts and their distribution."""
+
+import operator
+from typing import Dict, List, Tuple
+
+from medmodels.medrecord import MedRecord
+from medmodels.medrecord.medrecord import EdgesDirected
+from medmodels.medrecord.types import Group, NodeIndex
+
+
+def count_concept_connections(
+    medrecord: MedRecord, concept: Group, cohort: Group
+) -> Dict[NodeIndex, int]:
+    """Count for each concept node how often it is connected to the cohort group.
+
+    Args:
+        medrecord (MedRecord): Medrecord containing concept and cohort groups and with
+            nodes and also edges connecting these to groups.
+        concept (Group): Concept, for example 'Diagnoses', 'Procedure' or 'Medication'.
+        cohort (Group): Cohort group, for example 'Patients'.
+
+    Returns:
+        Dict[NodeIndex, int]: Dictionary with the node index for the concept and the
+            count for the connecting edges.
+    """
+    concept_counts = {}
+
+    concept_nodes = medrecord.group(concept)["nodes"]
+
+    for concept_node in concept_nodes:
+        concept_counts[concept_node] = len(
+            medrecord.edges_connecting(
+                concept_node,
+                medrecord.group(cohort)["nodes"],
+                directed=EdgesDirected.UNDIRECTED,
+            )
+        )
```

Comment:
Optimization problem found when using edges_connecting. Querying for those edges is apparently 100x faster.
```suggestion
    concept_nodes = medrecord.group(concept)["nodes"]

    def test_one(edge: EdgeOperand):
        edge.source_node().in_group(cohort)
        edge.target_node().index().equal_to(concept_node)

    def test_two(edge: EdgeOperand):
        edge.target_node().in_group(cohort)
        edge.source_node().index().equal_to(concept_node)

    for concept_node in tqdm(concept_nodes, desc="Getting concept counts"):
        concept_counts[concept_node] = len(
            medrecord.select_edges(lambda edge: edge.either_or(test_one, test_two))
        )
```

Reply:
Please adequate typing and docstrings of the query function accordingly. @JabobKrauskopf we should probably look into this mismatch in efficiency. Although I do not know if this could be someone implemented inside Rust, since I also use it in MTGAN and it takes a lot of time.